ChangeLog.d/improve_x509_cert_writing_serial_number_management.txt

```diff
@@ -0,0 +1,10 @@
+Bugfix
```

Comment:
Ah, good point, that's indeed a bug fix, as we were previously not compliant. So, this part will need to be backported to 2.28. Also, we should have a non-regression test. For development, that means checking that both the old and the new function reject overlong serials, and for 2.28 obviously on the old.

I think the simplest way to go about it is to add a new test function that just does invalid argument testing of `set_serial()`.